### PR TITLE
enhancement/request-retransmission

### LINKED ISSUE #23: Retransmitting REQUEST messages
Currently we don't retransmission `REQUEST`s this could lead to issues when assuming mostly offline nodes. To solve for this all `REQUEST` messages should be added to a queue and retransmissioned if no corresponding message was received after x epochs. The logic of this would be similar to how the retransmission of `OFFER`s works.

### PR DESCRIPTION
**Pull request by decanus**
closes #23

### CODE REVIEW

**Comment by decanus** [comment]
@oskarth, this is a rough implementation and not final yet, but opinion would be welcome already

**Comment by decanus** [comment]
Fun fact: this brings back race conditions

File: node.go [hunk 1]
```diff
@@ -85,8 +85,10 @@ func (n *Node) AppendMessage(group GroupID, data []byte) (MessageID, error) {
 					continue
 				}
 
-				s := state{}
-				s.SendEpoch = n.epoch + 1
+				s := state{
+					Type: OFFER,
```
Comment (oskarth) [inline node.go]: Why is OFFER singled out here if it is about requests?

File: node.go [hunk 2]
```diff
@@ -176,11 +183,14 @@ func (n *Node) onOffer(group GroupID, sender PeerId, msg Offer) [][]byte {
 			continue
 		}
 
-		r = append(r, raw)
+		s := state{
+			Type: REQUEST,
```
Comment (oskarth) [inline node.go]: Oh nevermind, it is here

**Comment by decanus** [comment]
We should run simulations to see what effect this has, also should `REQUEST` and `OFFER` next epoch calculation be the same?

**Review by oskarth — APPROVED** [review]

**Pull request merged**